[Refactor] UserType in Notification Classes

johnxnguyen opened the pull request:
## What's new in this PR?

This PR is part of an ongoing effort to all public references that leak concrete user objects. See the architecture issues for more info: https://github.com/wireapp/ios-architecture/issues/89

Public exposure of `ZMUser` has been removed from the following cluster of files:

- `NotificationUserInfo.swift`
- `ZMLocalNotification.swift`
- `LocalNotificationType+Localization.swift`
- `LocalNotificationDispatcher+Calling.swift`

## Note

I came across a lot of unnecessary public access modifiers, so I've removed those too to harden the SE boundary.

typfel commented on Source/Notifications/Push notifications/LocalNotificationDispatcher+Calling.swift:
```diff
@@ -20,25 +20,26 @@ import Foundation
-    func process(callState: CallState, in conversation: ZMConversation, caller: ZMUser) {
-        // missed call notification are handled separately
-        // but if call was answered elsewhere then proceed
+    func process(callState: CallState, in conversation: ZMConversation, caller: UserType) {
```
Isn't this an internal where it's ok to use the concrete `ZMUser` type

typfel commented on Source/Notifications/Push notifications/LocalNotificationDispatcher+Calling.swift:
```diff
@@ -20,25 +20,26 @@ import Foundation
     }
     
-    func processMissedCall(in conversation: ZMConversation, caller: ZMUser) {
+    func processMissedCall(in conversation: ZMConversation, caller: UserType) {
```
Also internal?

johnxnguyen commented:
@typfel I've reverted the commits you reviewed and instead removed public access. I also removed a lot of redundant public access in related classes (but not strictly necessary for the `ZMUser` type leakage.

johnxnguyen commented:
@zenkins release major
